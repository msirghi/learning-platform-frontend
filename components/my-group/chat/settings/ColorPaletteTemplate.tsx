import React from 'react';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import Radio from '@material-ui/core/Radio';

type Props = {
  color: string;
  checked: boolean;
  onChange: () => void;
};

export const ColorPaletteTemplate: React.FC<Props> = ({ onChange, color, checked }) => {
  return (
    <div className={styles.colorPaletteTemplate} style={{ backgroundColor: color }}>
      <div className={styles.firstMessage} />
      <div className={styles.secondMessage} />
      <div className={styles.radionButton}>
        <Radio checked={checked} color='primary' onChange={() => onChange()} />
      </div>
    </div>
  );
};
