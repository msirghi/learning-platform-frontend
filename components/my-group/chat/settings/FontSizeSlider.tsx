import { Slider } from '@material-ui/core';
import React from 'react';

const marks = [
  {
    value: 10,
    label: '10'
  },
  {
    value: 12,
    label: '12'
  },
  {
    value: 14,
    label: '14'
  },
  {
    value: 16,
    label: '16'
  }
];

type Props = {
  handleFontChange: (val: number) => void;
  value?: number;
};

export const FontSizeSlider: React.FC<Props> = ({ value, handleFontChange }) => {
  return (
    <Slider
      value={value ?? 14}
      aria-labelledby='discrete-slider-custom'
      step={1}
      onChange={(_, val) => handleFontChange(val as number)}
      valueLabelDisplay='auto'
      marks={marks}
      max={16}
      min={10}
    />
  );
};
