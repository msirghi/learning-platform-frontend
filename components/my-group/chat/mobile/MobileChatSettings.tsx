import React from 'react';
import { ColorPaletteTemplate } from '../settings/ColorPaletteTemplate';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import { chatThemes } from '../../../common/utils/chatThemes';
import { FontSizeSlider } from '../settings/FontSizeSlider';
import { Button } from '@material-ui/core';
import { ChatThemes } from '../../../../common/enums';

type Props = {
  currentColor: string;
  chatFontSize: number;
  handleThemeChange: (pallete: ChatThemes) => void;
  handleFontChange: (val: number) => void;
  toggleSettings: () => void;
};

export const MobileChatSettings: React.FC<Props> = ({
  currentColor,
  chatFontSize,
  handleThemeChange,
  handleFontChange,
  toggleSettings
}) => {
  return (
    <div className={styles.mobileChatSettingsContainer}>
      <div className={styles.mainTitle}>Chat settings</div>
      <div className={styles.themeTitle}>Theme</div>
      <div className={styles.themeSubtitle}>Select a theme for your chat</div>
      <div className={styles.mobileChatPalettes}>
        {chatThemes.map(({ color }) => {
          return (
            <div className={styles.theme} key={color}>
              <ColorPaletteTemplate
                onChange={() => handleThemeChange(color as ChatThemes)}
                color={color}
                checked={color === currentColor}
              />
            </div>
          );
        })}
      </div>

      <div className={styles.fontSizeTitle}>Font size</div>
      <div className={styles.fontSizeSubtitle}>Adjust the chat font size</div>
      <FontSizeSlider value={chatFontSize} handleFontChange={(val) => handleFontChange(val)} />

      <div className={styles.applyButtonContainer}>
        <Button color='primary' variant='contained' fullWidth onClick={() => toggleSettings()}>
          Apply
        </Button>
      </div>
    </div>
  );
};
