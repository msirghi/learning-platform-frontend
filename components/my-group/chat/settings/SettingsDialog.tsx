import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { ColorPaletteTemplate } from './ColorPaletteTemplate';
import { chatThemes } from '../../../common/utils/chatThemes';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import { FontSizeSlider } from './FontSizeSlider';
import { useTranslation } from '../../../../i18n';

type Props = {
  open: boolean;
  handleClose: () => void;
  handlePaletteChange: (pallete: string) => void;
  currentColor: string;
  chatFontSize: number;
  handleFontChange: (val: number) => void;
};

export const SettingsDialog: React.FC<Props> = ({
  currentColor,
  open,
  handleClose,
  handlePaletteChange,
  chatFontSize,
  handleFontChange
}) => {
  const { t } = useTranslation();

  return (
    <div>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby='alert-dialog-title'
        aria-describedby='alert-dialog-description'
      >
        <DialogTitle id='alert-dialog-title'>{'Chat settings'}</DialogTitle>
        <DialogContent>
          <span>{t('myGroup:themesTitle')}</span>
          <div className={styles.chatThemesContainer}>
            {chatThemes.map(({ color }) => {
              return (
                <div className={styles.theme} key={color}>
                  <ColorPaletteTemplate
                    onChange={() => handlePaletteChange(color)}
                    color={color}
                    checked={color === currentColor}
                  />
                </div>
              );
            })}
          </div>

          <span>{t('myGroup:fontSizeTitle')}</span>
          <FontSizeSlider value={chatFontSize} handleFontChange={handleFontChange} />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose} color='primary'>
            {t('myGroup:close')}
          </Button>
          <Button onClick={handleClose} color='primary' autoFocus>
            {t('myGroup:apply')}
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
};
