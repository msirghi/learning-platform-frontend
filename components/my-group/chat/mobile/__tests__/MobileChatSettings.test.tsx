import { Button } from '@material-ui/core';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ColorPaletteTemplate } from '../../settings/ColorPaletteTemplate';
import { FontSizeSlider } from '../../settings/FontSizeSlider';
import { MobileChatSettings } from '../MobileChatSettings';

describe('MobileChatSettings', () => {
  const props = {
    currentColor: '#fff',
    chatFontSize: 14,
    handleThemeChange: jest.fn(),
    handleFontChange: jest.fn(),
    toggleSettings: jest.fn()
  };

  it('should match the snapshot when all props are provided', () => {
    const wrapper = shallow(<MobileChatSettings {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should fire prop function on palette change', () => {
    const spy = jest.fn();
    const wrapper = shallow(<MobileChatSettings {...props} handleThemeChange={spy} />);

    const paletteTemplates = wrapper.find(ColorPaletteTemplate);
    expect(paletteTemplates.length > 1).toBeTruthy();

    paletteTemplates.at(0).props().onChange();
    expect(spy).toBeCalled();
  });

  it('should fire prop function on font size change', () => {
    const spy = jest.fn((data) => data);
    const wrapper = shallow(<MobileChatSettings {...props} handleFontChange={spy} />);

    const slider = wrapper.find(FontSizeSlider);
    expect(slider).toHaveLength(1);
    slider.props().handleFontChange(14);
    expect(spy).toBeCalled();
    expect(spy).toHaveBeenCalledWith(14);
  });

  it('should fire prop function on apply button click', () => {
    const spy = jest.fn();
    const wrapper = shallow(<MobileChatSettings {...props} toggleSettings={spy} />);

    const button = wrapper.find(Button);
    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(spy).toBeCalled();
  });
});
