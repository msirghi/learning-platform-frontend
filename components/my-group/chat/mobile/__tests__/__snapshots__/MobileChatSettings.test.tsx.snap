// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileChatSettings should match the snapshot when all props are provided 1`] = `
<div
  className="mobileChatSettingsContainer"
>
  <div
    className="mainTitle"
  >
    Chat settings
  </div>
  <div
    className="themeTitle"
  >
    Theme
  </div>
  <div
    className="themeSubtitle"
  >
    Select a theme for your chat
  </div>
  <div
    className="mobileChatPalettes"
  >
    <div
      className="theme"
      key="#fafafa"
    >
      <ColorPaletteTemplate
        checked={false}
        color="#fafafa"
        onChange={[Function]}
      />
    </div>
    <div
      className="theme"
      key="aliceblue"
    >
      <ColorPaletteTemplate
        checked={false}
        color="aliceblue"
        onChange={[Function]}
      />
    </div>
    <div
      className="theme"
      key="grey"
    >
      <ColorPaletteTemplate
        checked={false}
        color="grey"
        onChange={[Function]}
      />
    </div>
    <div
      className="theme"
      key="lightblue"
    >
      <ColorPaletteTemplate
        checked={false}
        color="lightblue"
        onChange={[Function]}
      />
    </div>
    <div
      className="theme"
      key="wheat"
    >
      <ColorPaletteTemplate
        checked={false}
        color="wheat"
        onChange={[Function]}
      />
    </div>
  </div>
  <div
    className="fontSizeTitle"
  >
    Font size
  </div>
  <div
    className="fontSizeSubtitle"
  >
    Adjust the chat font size
  </div>
  <FontSizeSlider
    handleFontChange={[Function]}
    value={14}
  />
  <div
    className="applyButtonContainer"
  >
    <WithStyles(ForwardRef(Button))
      color="primary"
      fullWidth={true}
      onClick={[Function]}
      variant="contained"
    >
      Apply
    </WithStyles(ForwardRef(Button))>
  </div>
</div>
`;
